flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcFilterPushdownPreparedStatementVisitor.java

```diff
@@ -255,4 +175,91 @@ public Optional<ParameterizedPredicate> visit(FieldReferenceExpression fieldRefe
     protected Optional<ParameterizedPredicate> defaultMethod(Expression expression) {
         return Optional.empty();
     }
+
+    /**
+     * Enum to map {@code BuiltInFunctionDefinitions} to required info for further push down
+     * processing with {@code JdbcFilterPushdownPreparedStatementVisitor}.
+     */
+    enum FilterPushdownFunction {
+        EQUALS(BuiltInFunctionDefinitions.EQUALS, "="),
+        LESS_THAN(BuiltInFunctionDefinitions.LESS_THAN, "<"),
+        GREATER_THAN(BuiltInFunctionDefinitions.GREATER_THAN, ">"),
+        LESS_THAN_OR_EQUAL(BuiltInFunctionDefinitions.LESS_THAN_OR_EQUAL, "<="),
+        GREATER_THAN_OR_EQUAL(BuiltInFunctionDefinitions.GREATER_THAN_OR_EQUAL, ">="),
+        NOT_EQUALS(BuiltInFunctionDefinitions.NOT_EQUALS, "<>"),
+        OR(BuiltInFunctionDefinitions.OR, "OR"),
+        AND(BuiltInFunctionDefinitions.AND, "AND"),
+        LIKE(BuiltInFunctionDefinitions.LIKE, "LIKE"),
+        IS_NULL(BuiltInFunctionDefinitions.IS_NULL, "IS NULL", true, true),
+        IS_NOT_NULL(BuiltInFunctionDefinitions.IS_NOT_NULL, "IS NOT NULL", true, true);
+
+        private final BuiltInFunctionDefinition function;
+        private final String functionSql;
+        private final boolean unaryOperation;
+        private final boolean operatorOnTheLeft;
+
+        private static final Map<FunctionDefinition, FilterPushdownFunction>
+                BUILT_IN_FUNCTION_2_FILTER_FUNCTION_MAP =
+                        Arrays.stream(values())
+                                .collect(Collectors.toMap(t -> t.function, Function.identity()));
+
+        FilterPushdownFunction(
+                BuiltInFunctionDefinition function,
+                String functionSql,
+                boolean unaryOperation,
+                boolean operatorOnTheLeft) {
+            this.function = function;
+            this.functionSql = functionSql;
+            this.unaryOperation = unaryOperation;
+            this.operatorOnTheLeft = operatorOnTheLeft;
+        }
+
+        FilterPushdownFunction(BuiltInFunctionDefinition function, String functionSql) {
+            this(function, functionSql, false, false);
+        }
+
+        Optional<ParameterizedPredicate> apply(
+                JdbcFilterPushdownPreparedStatementVisitor visitor, List<ResolvedExpression> args) {
+            if (unaryOperation) {
+                return visitor.renderUnaryOperator(functionSql, args.get(0), operatorOnTheLeft);
+            } else {
+                return visitor.renderBinaryOperator(functionSql, args);
+            }
+        }
+
+        static FilterPushdownFunction valueOf(FunctionDefinition builtInFunctionDefinition) {
+            return BUILT_IN_FUNCTION_2_FILTER_FUNCTION_MAP.get(builtInFunctionDefinition);
+        }
+
+        @VisibleForTesting
+        static Set<String> getBinaryOperators() {
```

Comment:
This method seems identical to getUnaryOperators. Also I can't spot where it is used

Reply:
they are not 100% same, the diff is in `filter` arg
yep, looks like they could be dropped since they are not used, good catch